src/main/webapp/app/exercises/programming/manage/update/update-components/programming-exercise-difficulty.component.html

```diff
@@ -50,6 +59,21 @@
                     }
                 </label>
             </div>
+            <div class="form-check" style="margin-left: 20px">
+                <label [class]="'form-check-label'" [ngStyle]="shouldDisableReadOnlyMode() ? { color: 'grey' } : {}" for="field_allowOnlineEditorReadOnlyMode">
+                    <input
+                        class="form-check-input"
+                        type="checkbox"
+                        name="allowOnlineEditorReadOnlyMode"
+                        id="field_allowOnlineEditorReadOnlyMode"
+                        [(ngModel)]="programmingExercise.allowOnlineEditorReadOnlyMode"
+                        [disabled]="shouldDisableReadOnlyMode()"
```

Comment:
No function-calls in templates. Consider reworking the logic or using Signals.
See the general client guidelines for more information